src/snowflake/connector/cursor.py

```diff
@@ -76,16 +77,16 @@
     from .connection import SnowflakeConnection
     from .file_transfer_agent import SnowflakeProgressPercentage
     from .result_batch import ResultBatch
+    from pyarrow import Table
 
 T = TypeVar("T", bound=collections.abc.Sequence)
 
 logger = getLogger(__name__)
 
-if installed_pandas:
-    from pyarrow import Table
-else:
+
+if not util.find_spec("pyarrow"):
```

Comment:
a check similar to this is done in snowflake.connector.options which tells if pandas/pyarrow are installed.

Reply:
Thanks. I initially didn't want to use it, since it's called `installed_pandas` and I couldn't see a `installed_pyarrow`. But I realise that it in the function it comes from, it will only evaluate to `True` if both pandas and pyarrow are installed. I decided to improve the warning a bit, since if the user has pyarrow but not pandas installed, the warning would still show.